### DISCUSSION TITLE
Make the popup open file for input and include configrarable for own commands

### DISCUSSION THREAD

**Discussion by obw** [texstudio-org/texstudio#3291] [Ideas]
I'm working on a steampunk Story and have there some complex Layouts, on of it is, that I want to have a graphical marker after each scene.

Also for the streamlining of working with SVN, I split every chapter into an own file!

The defined command which I have written, looks like this:

`\newcommand{\loadScene}[1]{\input{./kapitel/#1} \splitLine }`

The command ` \splitLine` loads a little graphics, which marks the End of a scene.

I think something like a comma-seperated list in the config dialog where I can simply say:

own include commands: loadScene, loadSomethingElse

regards

**Comment by obw** [comment]
Another thing what I have become aware just a moment ago, the things like the TODO-parser and outline, don't know what I'm doing!

this should be solved in a second step!

**Comment by sunderme** [comment]
have a look at the manual, especially how to write a cwl file.

**Reply by obw** [reply]
Nice, it's often to know where to look!

in `~/.config/texstudio/completion/user` a File with the name `mylayout.cwl`

and the following content works like a charm, after activating it in `Options` -> `TeXstudio config` -> `complition`!

\loadScene{file}#i

Something like that should be better documented, easier to find! I mean!

**Comment by obw** [comment]
Also, the second Problem is still existent! The TODOS which I insert in the Scenes don't show up in the global TODO List!
